Modules/rating/Controller/RatingController.php

```diff
@@ -0,0 +1,348 @@
+                $limit_date = strtotime($this->request->getParameter('limit_date'));
+            }
+
+            $new_campaign = $cm->set(
+                $id_space,
+                $id_campaign,
+                $from_date,
+                $to_date,
+                $limit_date,
+                $this->request->getParameter('message'),
+            );
+            if($id_campaign == 0 && $new_campaign) {
+                Events::send([
+                    "action" => Events::ACTION_RATING_CAMPAIGN_NEW,
+                    "space" => ["id" => intval($id_space)],
+                    "campaign" => ["id" => intval($new_campaign)]
+                ]);
+            }
+
+            $_SESSION['flashClass'] = 'success';
+            return $this->redirect('rating/'.$id_space, [], ['campaign' => ['id' => $new_campaign]]);
+        }
+
+        $r = new Rating();
+        $booking_ratings = [];
+        $projects_ratings = [];
+        $global_bookings = [];
+        $global_projects = [];
+        $total = [];
+        
+
+        if($id_campaign) {
+            $booking_ratings = $r->list($id_space, 'booking', campaign:$id_campaign);
+            for($i=0;$i<count($booking_ratings);$i++){
+                $booking_ratings[$i]['rate'] = intval($booking_ratings[$i]['rate']);
+                if(!$booking_ratings[$i]['login'] || $booking_ratings[$i]['anon']) {
+                    $booking_ratings[$i]['login'] = '';
+                }
+            }
+    
+
+
+            $projects_ratings = $r->list($id_space, 'projects', campaign:$id_campaign);
+            for($i=0;$i<count($projects_ratings);$i++){
+                $projects_ratings[$i]['rate'] = intval($projects_ratings[$i]['rate']);
+                if(!$projects_ratings[$i]['login'] || $projects_ratings[$i]['anon']) {
+                    $projects_ratings[$i]['login'] = '';
+                }
+            }
+
+
+            $global_stats = $r->stat($id_space, $id_campaign);
+
+            
+            foreach($global_stats as $g) {
+                if($g['module'] == 'booking') {
+                    $g['rate'] = round($g['rate']);
+                    $global_bookings[] = $g;
+                }
+            }
+
+            foreach($global_stats as $g) {
+                if($g['module'] == 'projects') {
+                    $g['rate'] = round($g['rate']);
+                    $global_projects[] = $g;
+                }
+            }
+
+            $total_stats = $r->statGlobal($id_space, $id_campaign);
+            foreach ($total_stats as $stat) {
+                $stat['rate'] = round($stat['rate']);
+                $total[$stat['module']] = $stat;
+            }
+        }
+
+
+        return $this->render(['data' => ['answers' => $answers, 'total' => $total, 'global_projects' => $global_projects, 'global_bookings' => $global_bookings, 'campaign' => $data, 'bookings' => $booking_ratings, 'projects' => $projects_ratings], 'lang' => $lang, 'form' => $form->getHtml($lang)]);
```

Comment:
$answers only defined i a else condition, line 129. Consequently not defined under certain circumstances.